include/zephyr/drivers/wifi/nrfwifi/off_raw_tx/off_raw_tx_api.h

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/** @file
+ *
+ * @addtogroup nrf70_off_raw_tx_api nRF70 Offloaded raw TX API
+ * @{
+ *
+ * @brief File containing API's for the Offloaded raw TX feature.
+ */
+
+#ifndef INCLUDE_ZEPHYR_DRIVERS_OFF_RAW_TX_API_H_
+#define INCLUDE_ZEPHYR_DRIVERS_OFF_RAW_TX_API_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+#include "osal_api.h"
+
+#define NRF_WIFI_OFF_RAW_TX_FRAME_SIZE_MIN 26
+#define NRF_WIFI_OFF_RAW_TX_FRAME_SIZE_MAX 600
+
+/**
+ * @brief- Transmission rates
+ * Rate to be used for transmitting a packet.
+ */
+enum nrf_wifi_off_raw_tx_rate {
+	/** 1 Mbps */
+	RATE_1M,
+	/** 2 Mbps */
+	RATE_2M,
+	/** 5.5 Mbps */
+	RATE_5_5M,
+	/** 11 Mbps */
+	RATE_11M,
+	/** 6 Mbps */
+	RATE_6M,
+	/** 9 Mbps */
+	RATE_9M,
+	/** 12 Mbps */
+	RATE_12M,
+	/** 18 Mbps */
+	RATE_18M,
+	/** 24 Mbps */
+	RATE_24M,
+	/** 36 Mbps */
+	RATE_36M,
+	/** 48 Mbps */
+	RATE_48M,
+	/** 54 Mbps */
+	RATE_54M,
+	/** MCS 0 */
+	RATE_MCS0,
+	/** MCS 1 */
+	RATE_MCS1,
+	/** MCS 2 */
+	RATE_MCS2,
+	/** MCS 3 */
+	RATE_MCS3,
+	/** MCS 4 */
+	RATE_MCS4,
+	/** MCS 5 */
+	RATE_MCS5,
+	/** MCS 6 */
+	RATE_MCS6,
+	/** MCS 7 */
+	RATE_MCS7,
+	/** Invalid rate */
+	RATE_MAX
+};
+
+
+/**
+ * @brief- HE guard interval value
+ * Value of the guard interval to be used between symbols when transmitting using HE.
+ */
+enum nrf_wifi_off_raw_tx_he_gi {
+	/** 800 ns */
+	HE_GI_800NS,
+	/** 1600 ns */
+	HE_GI_1600NS,
+	/** 3200 ns */
+	HE_GI_3200NS,
+	/** Invalid value */
+	HE_GI_MAX
+};
+
+
+/**
+ * @brief- HE long training field duration
+ * Value of the long training field duration to be used when transmitting using HE.
+ */
+enum nrf_wifi_off_raw_tx_he_ltf {
+	/** 3.2us */
+	HE_LTF_3200NS,
+	/** 6.4us */
+	HE_LTF_6400NS,
+	/** 12.8us */
+	HE_LTF_12800NS,
+	/** Invalid value */
+	HE_LTF_MAX
+};
+
+/**
+ * @brief- Throughput mode
+ * Throughput mode to be used for transmitting the packet.
+ */
+enum nrf_wifi_off_raw_tx_tput_mode {
+	/** Legacy mode */
+	TPUT_MODE_LEGACY,
+	/** High Throughput mode (11n) */
+	TPUT_MODE_HT,
+	/** Very high throughput mode (11ac) */
+	TPUT_MODE_VHT,
+	/** HE SU mode */
+	TPUT_MODE_HE_SU,
+	/** HE ER SU mode */
+	TPUT_MODE_HE_ER_SU,
+	/** HE TB mode */
+	TPUT_MODE_HE_TB,
+	/** Highest throughput mode currently defined */
+	TPUT_MODE_MAX
+};
+
+/**
+ * @brief This structure defines the Offloaded raw tx debug statistics.
+ *
+ */
+struct nrf_wifi_off_raw_tx_stats {
+	/** Number of packets sent */
+	unsigned int off_raw_tx_pkt_sent;
+};
+
+/**
+ * @brief- Configuration parameters for offloaded raw TX
+ * Parameters which can be used to configure the offloaded raw TX operation.
+ */
+struct nrf_wifi_off_raw_tx_conf {
+	/** Time interval (in microseconds) between transmissions */
+	unsigned int period_us;
+	/** Transmit power in dBm (0 to 20) */
```

Comment:
what about regulatory settings here? How can a user configure the regulatory domain?